spirv_msl.cpp

```diff
@@ -4991,9 +5140,33 @@ void CompilerMSL::emit_store_statement(uint32_t lhs_expression, uint32_t rhs_exp
 
 	bool transpose = lhs_e && lhs_e->need_transpose;
 
-	if (has_decoration(lhs_expression, DecorationBuiltIn) &&
-	    BuiltIn(get_decoration(lhs_expression, DecorationBuiltIn)) == BuiltInSampleMask &&
-	    is_array(type))
+	// Meshlet indices
+	if (lhs_e != nullptr && lhs_e->loaded_from == builtin_mesh_primitive_indices_id &&
```

Comment:
This is not the right place to do it. Leaf functions should receive a plain `threadgroup uint3 gl_Indices[MaxPrimitives]` array. The lowering to set_index needs to happen in the wrapped main.

Pretty sure you can read the content of that array in SPIR-V and this code would break that.

Reply:
going thru spec again: yep, my understanding that all `out` variables are write only is not correct.